### PR TITLE
T2 automation-CEPH-10753-Test deleting a zone

### PR DESCRIPTION
**Pull request by anrao19**
configure an additional zone in one of the sites and update the period. 
Delete the zone and update. --> shoule be successful.
log: 
53: http://magna002.ceph.redhat.com/ceph-qe-logs/Anuchaithra/test_zone_deletion.console.log
61: http://magna002.ceph.redhat.com/ceph-qe-logs/Anuchaithra/test_zone_deletion_61.console.log

### CODE REVIEW

File: rgw/v2/tests/s3_swift/test_check_sharding_enabled.py [hunk 1]
```diff
@@ -94,6 +94,34 @@ def test_exec(config, ssh_con):
         else:
             raise TestExecError("sharding has not enabled already")
 
+    if config.test_ops.get("zone_delete", False) is True:
+        log.info("Test zone deletion")
+        op = utils.exec_shell_cmd("radosgw-admin sync status")
+        lst = list(op.split("\n"))
```
Comment (ckulal) [inline rgw/v2/tests/s3_swift/test_check_sharding_enabled.py:100]: please use the meaningful variable name to understand easily.

File: rgw/v2/tests/s3_swift/test_check_sharding_enabled.py [hunk 2]
```diff
@@ -94,6 +94,34 @@ def test_exec(config, ssh_con):
...
+        for l in lst:
```
[the remark this answers is not in the document]
Comment (anrao19) [reply rgw/v2/tests/s3_swift/test_check_sharding_enabled.py]: done

File: rgw/v2/tests/s3_swift/test_check_sharding_enabled.py [hunk 3]
```diff
@@ -94,6 +94,34 @@ def test_exec(config, ssh_con):
...
+        )
+        utils.exec_shell_cmd("radosgw-admin period update --commit")
+        zone_list = json.loads(utils.exec_shell_cmd("radosgw-admin zone list"))
+        if not zone_name in zone_list["zones"]:
+            raise TestExecError(f"Zone {zone_name} does not exist")
+        log.info(f"Delete zone {zone_name} from zone group {zonegroup_name}")
+        utils.exec_shell_cmd(
+            f"radosgw-admin zonegroup remove --rgw-zonegroup {zonegroup_name} --rgw-zone {zone_name}"
+        )
+        utils.exec_shell_cmd(f"radosgw-admin zone delete --rgw-zone {zone_name}")
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/test_check_sharding_enabled.py:120]: @anrao19 , could you please add period update commit after zonegroup remove and zone delete

Comment (anrao19) [reply rgw/v2/tests/s3_swift/test_check_sharding_enabled.py:120]: @hmaheswa, updated the same
log: http://magna002.ceph.redhat.com/ceph-qe-logs/Anuchaithra/test_zone_deletion_61.console.log

**Review by hmaheswa — APPROVED** [review]

**Review by ckulal — COMMENTED** [review]
looks good

File: rgw/v2/tests/s3_swift/test_check_sharding_enabled.py [hunk 4]
```diff
@@ -94,6 +94,35 @@ def test_exec(config, ssh_con):
...
+    if config.test_ops.get("zone_delete", False) is True:
+        log.info("Test zone deletion")
+        op = utils.exec_shell_cmd("radosgw-admin sync status")
+        lines = list(op.split("\n"))
+        for line in lines:
+            if "realm" in line:
+                realm = line[line.find("(") + 1 : line.find(")")]
+            if "zonegroup" in line:
+                zonegroup_name = line[line.find("(") + 1 : line.find(")")]
+                break
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/test_check_sharding_enabled.py:106]: @anrao19 , there is a function in utils to get the realm and source zone info https://github.com/red-hat-storage/ceph-qe-scripts/blob/master/rgw/v2/utils/utils.py#L636 , can we reuse it?

Comment (anrao19) [reply rgw/v2/tests/s3_swift/test_check_sharding_enabled.py:106]: Hi @mkasturi18 That method mentioned does not provide the zonegroup name 
source zone in this method is to check if its master or not

**Comment by TejasC88** [comment]
looks good.

**Review by TejasC88 — APPROVED** [review]

**Pull request merged**